pages/api/transactions/index.js

```diff
@@ -1,39 +1,62 @@
 import dbConnect from "@/db/connect";
 import Transaction from "@/db/models/Transaction";
+import { getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
+import { getToken } from "next-auth/jwt";
 
 export default async function handler(request, response) {
+  const session = await getServerSession(request, response, authOptions);
+  const token = await getToken({ req: request });
+  const userId = token?.sub;
+
+  console.log(token);
   await dbConnect();
+
   if (request.method === "GET") {
-    try {
-      const transactions = await Transaction.find();
+    /* const transactions = await Transaction.find();
+    response.status(200).json(transactions);
+    return; */
+
+    if (session) {
+      const transactions = await Transaction.find({ owner: token.email });
       response.status(200).json(transactions);
       return;
-    } catch (error) {
-      response.status(500).json({ message: "Error finding transactions" });
+    } else {
+      const transactions = await Transaction.find({ owner: "default" });
```

Comment:
Does the 'else' case make sense? This is if you have any default transactions for not loggedin users. This makes no sense for a money app in my eyes. You cannot even access the app without being logged in.